pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,499 @@
+	QueueName string
+}
+
+var listPodFlags = &listFlags{}
+
+// InitListFlags init list command flags.
+func InitListFlags(cmd *cobra.Command) {
+	util.InitFlags(cmd, &listPodFlags.CommonFlags)
+
+	cmd.Flags().StringVarP(&listPodFlags.QueueName, "queue", "q", "", "list pod with specified queue name")
+	cmd.Flags().StringVarP(&listPodFlags.JobName, "job", "j", "", "list pod with specified job name")
+	cmd.Flags().StringVarP(&listPodFlags.Namespace, "namespace", "n", "default", "the namespace of job")
+	cmd.Flags().BoolVarP(&listPodFlags.allNamespace, "all-namespaces", "", false, "list jobs in all namespaces")
+}
+
+// ListPods lists all pods details created by vcjob
+func ListPods(ctx context.Context) error {
+	config, err := util.BuildConfig(listPodFlags.Master, listPodFlags.Kubeconfig)
+	if err != nil {
+		return err
+	}
+	if listPodFlags.allNamespace {
+		listPodFlags.Namespace = ""
+	}
+
+	var pods corev1.PodList
+
+	// filter pods based on annotationKey and annotationValue
+	filterByAnnotationFunc := func(pods *corev1.PodList, annotationKey, annotationValue string) *corev1.PodList {
+		filteredPods := &corev1.PodList{}
+		for _, pod := range pods.Items {
+			if value, exist := pod.Annotations[annotationKey]; exist {
+				if value == annotationValue {
+					filteredPods.Items = append(filteredPods.Items, pod)
+				}
+			}
+		}
+		return filteredPods
+	}
+
+	// filter pods based on labelKey and labelValue
+	// isFastReturn indicates whether to return immediately if a match is found
+	filterByLabelFunc := func(pods *corev1.PodList, labelKey, labelValue string, isFastReturn bool) (*corev1.PodList, error) {
+		filteredPods := &corev1.PodList{}
+		for _, pod := range pods.Items {
+			if value, exist := pod.Labels[labelKey]; exist {
+				if value == labelValue {
+					filteredPods.Items = append(filteredPods.Items, pod)
+				} else {
+					if !isFastReturn {
+						continue
+					}
+					// return immediately if a match is found and isFastReturn is true
+					return filteredPods, fmt.Errorf("the input vcjob %s does not match the queue %s",
+						listPodFlags.JobName, listPodFlags.QueueName)
+				}
+			}
+		}
+		return filteredPods, nil
+	}
+
+	// if job name are specified, use job name to filter pods
+	// if the job is specified, it means that no matter whether the queue name is specified or not,
+	// we can just use the job name query. Because vcjob itself will specify a specific queue
+	if listPodFlags.JobName != "" {
+		labelSelector, err := createPodLabelSelectorByJobName(listPodFlags.JobName)
+		if err != nil {
+			return err
+		}
+		listVcjobPodsRes, err := listPodByLabel(ctx, config, labelSelector)
+		if err != nil {
+			return err
+		}
+		pods.Items = append(pods.Items, listVcjobPodsRes.Items...)
+		if listPodFlags.QueueName != "" {
+			// if queue is specified, check if the queue name used by the pod matches with the one passed in
+			_, err := filterByLabelFunc(listVcjobPodsRes, v1alpha1.QueueNameKey, listPodFlags.QueueName, true)
```

Comment:
There is no need to set a flag in filterByLabelFunc, which increased 
 code complexity, just compare pods.Items[0].queueName and listPodFlags.QueueName is ok.

Reply:
fixed